doc/rfcs/2025-11-30--InterProcessCommunication.md

```diff
@@ -0,0 +1,440 @@
+others. One example of this is shared memory, where the number of memory
+protection regions a microcontroller is capable of will limit the number of
+memory regions IPC can support. When possible, IPC mechanisms should scale
+their capability based on the microcontroller rather than reject a platform
+altogether.
+
+**Mechanisms should primarily be implemented in capsules.**
+Capsules should be used to implement IPC mechanisms, with as much of the
+functionality as possible provided within the capsule. This is valuable as
+capsules are easier to extend for downstream users and cannot use unsafe Rust.
+Some functionality by necessity will need to exist in the kernel. This
+functionality should be as minimal as possible and should be protected from
+access with capabilities.
+
+
+### Non-goals
+
+**Optimal performance.**
+Much of IPC research focuses on optimal performance to enable microkernel
+designs. Our focus is instead of client-server interactions between userspace
+applications which may not require the best possible performance. Where
+possible, performance should be valued, but we will weigh that value when in
+conflict with other issues such as usability. However, the IPC system should
+be extensible to enable performance-focused designs for downstream users with
+stricter performance requirements.
+
+**One perfect mechanism.**
+We do not believe that there is any single best IPC mechanism that would
+support all application needs. Instead, we focus on an ecosystem of capsule
+implementations, with some basic mechanisms provided initially and others added
+as-needed. Additional capsules should be creatable downstream which could
+revise the interface of existing mechanisms, optimize existing mechanisms for a
+particular use case, or provide altogether new mechanisms.
+
+**Peer-to-Peer Discovery and Initiation.**
+The primary focus of the IPC design is to support client-server interactions
+where clients discover servers and initiate a connection. The current belief is
+that most applications will have a clear differentiation between servers which
+have resources and clients which take actions to connect with them. As an
+example of how this is applied to designs, in the current design servers
+"discover" clients when the client first contacts it, usually via the
+Synchronous Mailbox. An alternative design where any process could discover and
+reach out to any other process is possible, but is not the current focus. Note
+that this does not restrain communication patterns after initiation. Both
+client requests and asynchronous server notifications are supported in
+communication mechanisms.
+
+
+## IPC Name Registry Capsule
+
+This capsule provides registration for services and discovery for clients.
+Upon discovery, it provides opaque process IDs which can be used to refer to
+processes in other IPC mechanisms.
+
+If validation of services or clients is desired, this capsule could perform
```

Comment:
What does "validation" mean here? Client authentication and/or authorization, or something else (like verifying the client process implements particular APIs)?